examples/lm1b/train.py

```diff
@@ -29,76 +29,49 @@
-  """Creates learning rate schedule.
+def rsqrt_schedule(
+    init_value: float,
+    shift: int = 0,
+):
+  """Applies a reverse square-root schedule.
+  
+  The reverse square root schedule is simply `lr = init_value / sqrt(step)`.
+  Args:
+    init_value: Base learning rate (before applying the rsqrt schedule).
+    shift: How many steps the rsqrt should be shifted. Shifting the rsqrt
+      schedule makes it less steep in the beginning (close to 0).
+  """
 
-  Interprets factors in the factors string which can consist of:
-  * constant: interpreted as the constant value,
-  * linear_warmup: interpreted as linear warmup until warmup_steps,
-  * rsqrt_decay: divide by square root of max(step, warmup_steps)
-  * rsqrt_normalized_decay: divide by square root of max(step/warmup_steps, 1)
-  * decay_every: Every k steps decay the learning rate by decay_factor.
-  * cosine_decay: Cyclic cosine decay, uses steps_per_cycle parameter.
+  def schedule(count):
+    return init_value * (count + shift)**-.5 * shift**.5
 
-  Args:
-    factors: string, factors separated by "*" that defines the schedule.
-    base_learning_rate: float, the starting constant for the lr schedule.
-    warmup_steps: int, how many steps to warm up for in the warmup schedule.
-    decay_factor: float, the amount to decay the learning rate by.
-    steps_per_decay: int, how often to decay the learning rate.
-    steps_per_cycle: int, steps per cycle when using cosine decay.
+  return schedule
 
-  Returns:
-    a function learning_rate(step): float -> {"learning_rate": float}, the
-    step-dependent lr.
-  """
-  factors = [n.strip() for n in factors.split("*")]
-
-  def step_fn(step):
-    """Step to learning rate function."""
-    ret = 1.0
-    for name in factors:
-      if name == "constant":
-        ret *= base_learning_rate
-      elif name == "linear_warmup":
-        ret *= jnp.minimum(1.0, step / warmup_steps)
-      elif name == "rsqrt_decay":
-        ret /= jnp.sqrt(jnp.maximum(step, warmup_steps))
-      elif name == "rsqrt_normalized_decay":
-        ret *= jnp.sqrt(warmup_steps)
-        ret /= jnp.sqrt(jnp.maximum(step, warmup_steps))
-      elif name == "decay_every":
-        ret *= (decay_factor**(step // steps_per_decay))
-      elif name == "cosine_decay":
-        progress = jnp.maximum(0.0,
-                               (step - warmup_steps) / float(steps_per_cycle))
-        ret *= jnp.maximum(0.0,
-                           0.5 * (1.0 + jnp.cos(jnp.pi * (progress % 1.0))))
-      else:
-        raise ValueError("Unknown factor %s." % name)
-    return jnp.asarray(ret, dtype=jnp.float32)
-
-  return step_fn
+def create_learning_rate_schedule(learning_rate: float, warmup_steps: int):
+  """Creates a rsqrt schedule with linear warmup."""
+  return optax.join_schedules([
+      optax.linear_schedule(
+          init_value=0, end_value=learning_rate, transition_steps=warmup_steps),
+      rsqrt_schedule(init_value=learning_rate, shift=warmup_steps),
```

Comment:
When using the `rsqrt_schedule` you probably also need to update the learning rate in the config. Please check with #1476 and see [this Colab](https://colab.corp.google.com/drive/1CngW6pCIHH4yppyqsqnTfKXvkvqUzmIt)

Reply:
Thanks! I see that you've changed default `config.learning_rate` from `0.0625` to `0.002` while decreasing the default batch size as well. I requested access for the colab. To be consistent in the end, what's the learning_rate and batch_size used for results above?

Reply:
(sorry gave you access)

right so in your case it would be

```
learning_rate = 0.05
warmup_steps = 1000
num_train_steps = 500_000

lrmax = learning_rate / (warmup_steps ** .5)
lrmax
```

So 0.0015811388300841897 or the 0.0016 that you propose LGTM